src/routes/resources/+page.svelte

```diff
@@ -2,21 +2,98 @@
 	import ResourceCard from 'components/resource/ResourceCard.svelte';
 	import Section from 'components/layout/Section.svelte';
 	import SeoMetaTags from 'components/layout/SeoMetaTags.svelte';
+	import { Search, Telescope } from '@lucide/svelte';
+	import { fade, fly } from 'svelte/transition';
+
 	let { data } = $props();
+
+	// Search State
+	let searchQuery = $state('');
+
+	// Derived filtered list based on search query
+	let filteredResources = $derived(
+		data.resources.filter((r) => {
+			const query = searchQuery.toLowerCase();
+			return (
+				r.title.toLowerCase().includes(query) || 
+				r.description.toLowerCase().includes(query)
+			);
+		})
+	);
 </script>
 
 <SeoMetaTags
-	title="ECSESS Resources - Academic, Technical, Involvement, etc."
-	description="ECSESS resource hub for everything relating to academic, technical, involvement, sustainability, equity, campus life, and more!"
+	title="ECSESS Resources - Academic, Technical, Involvement & More"
+	description="The central hub for all ECSESS student resources including academic rights, building access, financial bursaries, and student handbooks."
 	canonical={data.canonical}
 />
 
-<Section from="from-ecsess-black" to="to-ecsess-black" via="via-ecsess-800" direction="to-b">
-	<p class="page-title">Resources</p>
+<Section from="from-ecsess-black" to="to-ecsess-black" via="via-ecsess-900" direction="to-b">
+	
+	<div class="mb-12 text-center max-w-3xl mx-auto space-y-4">
+		<h1 class="page-title text-ecsess-100">Student Resources</h1>
+		<p class="text-ecsess-200 text-lg md:text-xl leading-relaxed">
+			Everything you need to navigate your engineering journey at McGill, curated by ECSESS.
+		</p>
+	</div>
+
+	<div class="w-full max-w-2xl mx-auto mb-12 relative z-10">
+		<div class="relative group">
+			<div class="absolute inset-y-0 left-0 pl-4 flex items-center pointer-events-none">
+				<Search class="h-6 w-6 text-ecsess-400 group-focus-within:text-ecsess-200 transition-colors" />
+			</div>
+			<input
+				type="text"
+				bind:value={searchQuery}
+				placeholder="Search for handbooks, forms, or guides..."
+				class="block w-full pl-12 pr-4 py-4 bg-ecsess-900/50 border-2 border-ecsess-700 rounded-2xl 
+					   text-ecsess-50 placeholder-ecsess-400 focus:outline-none focus:border-ecsess-400 
+					   focus:ring-4 focus:ring-ecsess-400/20 transition-all text-lg shadow-lg backdrop-blur-sm"
+			/>
```

Comment:
Change the order. The Search icon is completely blurred because the input box is put on top of it.

```suggestions
			<input
				type="text"
				bind:value={searchQuery}
				placeholder="Search for handbooks, forms, or guides..."
				class="block w-full pl-12 pr-4 py-4 bg-ecsess-900/50 border-2 border-ecsess-700 rounded-2xl 
					   text-ecsess-50 placeholder-ecsess-400 focus:outline-none focus:border-ecsess-400 
					   focus:ring-4 focus:ring-ecsess-400/20 transition-all text-lg shadow-lg backdrop-blur-sm"
			/>
			<div class="absolute inset-y-0 left-0 pl-4 flex items-center pointer-events-none">
				<Search class="h-6 w-6 text-ecsess-400 group-focus-within:text-ecsess-200 transition-colors" />
			</div>
```

Your original code:

<img width="463" height="113" alt="Image" src="https://github.com/user-attachments/assets/645bf838-b428-4271-bf9d-62c6f3244ac8" />

whereas it should be:

<img width="463" height="113" alt="Image" src="https://github.com/user-attachments/assets/ec288807-cc57-45c2-b9b7-27bcea70cbeb" />